### PR TITLE
Products on store page

### PR DESCRIPTION
**Pull request by CodeNinja500**
Added product list and search logic on Store-products-page.

Draft pull, because I am waiting on merging the previous branch.

### CODE REVIEW

**Review by gradzio — COMMENTED** [review]
LGTM

File: src/app/components/store-products/store-products.component.ts [hunk 1]
```diff
@@ -8,4 +19,67 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
...
+      logoUrl: store.logoUrl,
+      distanceInKm: Math.round(store.distanceInMeters / 100) / 10,
+      tags: store.tagIds.map((tagId) => tagsMap[tagId]?.name)
+    };
+  }
+
+  onStoreSearchSubmitted(storeSearch: FormGroup): void {
+    //for debugging
+    console.log('Searching for products with keyWord: ' + storeSearch.value.keyWord);
+    this._keyWordForSearchSubject.next(storeSearch.value.keyWord.toLowerCase());
```
Comment (gradzio) [inline src/app/components/store-products/store-products.component.ts]: try to use valueChanges instead of a submit

Comment (CodeNinja500) [reply src/app/components/store-products/store-products.component.ts]: I wanted to make it work on pressing Enter or clicking on the icon. But shure I can also do it on value changes :)

Comment (CodeNinja500) [reply src/app/components/store-products/store-products.component.ts]: Search logic changed :)

**Review by gradzio — APPROVED** [review]

File: src/app/components/store-products/store-products.component.ts (hunk 1 above)
Comment (gradzio) [reply src/app/components/store-products/store-products.component.ts]: ok i see, also good idea

**Pull request merged**